#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import ParameterGenerator, double_t, int_t, bool_t

gen = ParameterGenerator()

gen.add("carrot_distance", int_t, 0, "The number of points on the global plan at which the carrot maximaly positioned.", 1, 0, 100)

gen.add("rotation_accuracy", double_t, 0, "Accuracy of the orientation to the goal orientation.", 0.1, 0, 3.14)

gen.add("position_accuracy", double_t, 0, "Maximal distance to the goal position.", 0.05, 0, 1.0)

gen.add("max_rotation_vel", double_t, 0, "The absolute value of the maximum rotational velocity for the robot in rad/s.", 0.8, 0, 20.0)

gen.add("max_vel_x", double_t, 0, "The maximum x velocity for the robot in m/s.", 0.15, 0, 20.0)
gen.add("min_vel_x", double_t, 0, "The maximum x velocity for the robot in m/s.", 0.15, -20, 0)
gen.add("max_vel_y", double_t, 0, "The maximum x velocity for the robot in m/s.", 0.15, 0, 20.0)

gen.add("rotation_slow_down_factor", double_t, 0, "Factor which scales the minimal angle difference to the goal rotational velocity in rad/s", 0.2, 0.0, 1.0)

gen.add("translation_slow_down_factor", double_t, 0, "Factor which scales the difference in meters to the apprach velocity in m/s", 0.3, 0.0, 1.0)

gen.add("orient_to_goal_distance", double_t, 0, "Distance at which the robot preferes the goal orientaion over the path heading", 1, 0.0, 100.0)

exit(gen.generate("bitbots_local_planner", "bitbots_local_planner", "BBPlanner"))
